### PR TITLE
Improve docs for empty value in cache compression config

### PR DESCRIPTION
**Pull request by salvacorts**
**What this PR does / why we need it**:
Follow up PR for https://github.com/grafana/loki/pull/9535#discussion_r1218167670

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: docs/sources/configuration/_index.md [hunk 1]
```diff
@@ -798,8 +798,8 @@ index_stats_results_cache:
   # frontend.index-stats-results-cache
   [cache: <cache_config>]
 
-  # Use compression in results cache. Supported values are: 'snappy' and ''
-  # (disable compression).
+  # Use compression in cache. Supported values are: 'snappy' and ''. Default ''.
```
Comment (JStickler) [inline docs/sources/configuration/_index.md:801]:
```suggestion
  # Use compression in cache. The default is an empty value '', which disables compression.
```

**Review by JStickler — COMMENTED** [review]
[docs team]

File: docs/sources/configuration/_index.md [hunk 2]
```diff
@@ -798,8 +798,8 @@ index_stats_results_cache:
...
+  # An empty value (i.e. '') disables compression.
```
[the remark this answers is not in the document]
Comment (salvacorts) [reply docs/sources/configuration/_index.md]: Done

**Review by JStickler — APPROVED** [review]
[docs team] LGTM

**Pull request merged**